### PR TITLE
K8SPXC-734: Include namespace in maual recovery command

### PR DESCRIPTION
**Pull request by ExiaSR**
[K8SPXC-734](https://jira.percona.com/browse/K8SPXC-734) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

Make our life easier to run the `kubectl exec` command when recovering manually. No need to config the correct namespace in the current context or manually adding the `-n <namespace>` argument into the recovery command.

Using the downward API to inject the pod namespace as `MY_POD_NAMESPACE` environment variable

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scheduled-backup | passed |
| storage | failed |
| upgrade-haproxy | failed |
| haproxy | passed |
| scaling | passed |
| big-data | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| limits | failed |
| upgrade-proxysql | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| smart-update | passed |
| affinity | failed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |
| users | passed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/ee34055225e15056998bf471be9846eff5c9dc6a
image: `perconalab/percona-xtradb-cluster-operator:PR-864-ee340552`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| haproxy | passed |
| demand-backup | passed |
| pitr | passed |
| init-deploy | failed |
| demand-backup-encrypted-with-tls | passed |
| big-data | passed |
| scheduled-backup | passed |
| storage | failed |
| scaling | passed |
| upgrade-haproxy | failed |
| scaling-proxysql | passed |
| security-context | failed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| limits | failed |
| monitoring-2-0 | failed |
| upgrade-proxysql | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| smart-update | passed |
| upgrade-consistency | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/ee34055225e15056998bf471be9846eff5c9dc6a
image: `perconalab/percona-xtradb-cluster-operator:PR-864-ee340552`

File: pkg/pxc/app/statefulset/node.go [hunk 1]
```diff
@@ -136,6 +136,12 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaXt
 					SecretKeyRef: app.SecretKeySelector(secrets, "monitor"),
 				},
 			},
+			{
+				Name: "MY_POD_NAMESPACE",
```
Comment (qJkee) [inline pkg/pxc/app/statefulset/node.go:140]: better to name it as `POD_NAMESPACE`

**Review by qJkee — APPROVED** [review]

File: pkg/pxc/app/statefulset/node.go (hunk 1 above)
Comment (qJkee) [inline pkg/pxc/app/statefulset/node.go:140]: better to name it as `POD_NAMESPACE`

**Review by qJkee — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/a154cbd819b7e82c88e1eed5746fe4bcdcba546a
image: `perconalab/percona-xtradb-cluster-operator:PR-864-a154cbd8`

**Review by qJkee — DISMISSED** [review]

File: pkg/pxc/app/statefulset/node.go [hunk 2]
```diff
@@ -136,6 +136,12 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaXt
 					SecretKeyRef: app.SecretKeySelector(secrets, "monitor"),
 				},
 			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
```
Comment (delgod) [inline pkg/pxc/app/statefulset/node.go:141]: adding new vars adds unnecessary restarts of database during the update of the operator.
can you please take namespace from `/run/secrets/kubernetes.io/serviceaccount/namespace` file inside `build/pxc-entrypoint.sh` without adding any new env variables

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| upgrade-haproxy | failed |
| recreate | passed |
| haproxy | passed |
| scheduled-backup | passed |
| big-data | passed |
| scaling | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| scaling-proxysql | passed |
| init-deploy | failed |
| upgrade-proxysql | failed |
| security-context | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| limits | failed |
| operator-self-healing | failed |
| monitoring-2-0 | failed |
| smart-update | passed |
| pitr | passed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | failed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/a154cbd819b7e82c88e1eed5746fe4bcdcba546a
image: `perconalab/percona-xtradb-cluster-operator:PR-864-a154cbd8`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| upgrade-haproxy | passed |
| storage | passed |
| scheduled-backup | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| limits | passed |
| monitoring-2-0 | failed |
| smart-update | passed |
| operator-self-healing | failed |
| restore-to-encrypted-cluster | passed |
| affinity | passed |
| one-pod | passed |
| demand-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | failed |
| pitr | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/a9da0d6b445b2156b62e8604c5400a7e4748a196
image: `perconalab/percona-xtradb-cluster-operator:PR-864-a9da0d6b`

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| scaling | passed |
| scaling-proxysql | passed |
| big-data | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/864/commits/a9da0d6b445b2156b62e8604c5400a7e4748a196
image: `perconalab/percona-xtradb-cluster-operator:PR-864-a9da0d6b`

**Pull request merged**